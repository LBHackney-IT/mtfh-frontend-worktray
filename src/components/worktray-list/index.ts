export * from "./process-record";
export * from "./worktray-list";
export * from "./tenure-record";
