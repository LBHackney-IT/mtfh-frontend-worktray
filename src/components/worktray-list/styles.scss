@import "~lbh-frontend/lbh/base";
@import "~lbh-frontend/lbh/settings/colour-palette";

.process-record {
  &__item > .title {
    font-weight: bold;
    margin-top: 0;
  }

  &__item > .createdAt {
    margin-top: 0;
  }

  &__item {
    width: 200px;
    margin-top: 0;
    &.--processName {
      flex-grow: 1;
    }
    .lbh-tag {
      text-align: center;
      border-radius: 5px;
      width: 100%;
    }
    & > .--address {
      font-size: 16px;
      margin-top: 0;
      display: block;
      line-height: 22px;
    }
  }
}
