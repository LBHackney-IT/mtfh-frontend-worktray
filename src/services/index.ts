export { default as locale } from "./locale";
export { config } from "./config";
