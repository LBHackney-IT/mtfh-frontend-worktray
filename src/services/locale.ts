const locale = {
  title: "Your tasks",
  views: {
    worktray: {
      noWorktrayResults: "No tasks found",
    },
  },
  errors: {
    unableToFetchRecord: "There was a problem retrieving the record",
    unableToFetchRecordDescription:
      "Please try again. If the issue persists, please contact support.",
    unableToFindProcess: "Invalid Process",
    unableToFindProcessDescription: "We have been unable to find a process of this type.",
    unableToFindState: "Invalid State",
    unableToFindStateDescription: "We have been unable to process a state of this type.",
  },
  components: {
    filters: {
      applyFilters: "Apply filters",
      clearFilters: "Clear filters",
      selectAll: "Select All",
      removeAll: "Remove All",
    },
    pagination: {
      previous: "Previous",
      next: "Next",
      assistiveNavigation: (page: number): string => `Page ${page}`,
      searchResultsCounter: (page: number, pageSize: number, total: number): string =>
        total === 0
          ? "Showing 0 of 0 results"
          : `Showing ${page * pageSize - pageSize + 1}â€”${Math.min(
              page * pageSize,
              total,
            )} of ${total} results`,
    },
    controls: {
      timePeriodOptions: {
        "30": "Last 30 days",
        "60": "Last 60 days",
        "3": "Last 3 months",
        "6": "Last 6 months",
        "12": "Last 12 months",
      },
      limitOption: (value: number): string => `${value} items`,
    },
  },
};

export default locale;
