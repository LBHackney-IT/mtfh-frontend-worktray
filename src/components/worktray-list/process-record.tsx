import React from "react";
import { Link as RouterLink } from "react-router-dom";

import { addDays, differenceInDays, differenceInHours, parseISO } from "date-fns";

import { Link, Td, Text, Tr } from "@mtfh/common/lib/components";
import { formatDate } from "@mtfh/common/lib/utils";
import { IProcess, ProcessStateInfo } from "@mtfh/processes";

import { Process, ProcessName } from "../../types";

import "./styles.scss";

export interface ProcessRecordProps {
  person: {
    id?: string;
    fullName?: string;
  };
  property?: {
    id?: string;
    address?: string;
  };
  process: Process;
  processConfig: IProcess | undefined;
  simple?: boolean;
}

enum ProcessStates {
  PROCESS_CLOSED = "ProcessClosed",
  PROCESS_CANCELLED = "ProcessCancelled",
  PROCESS_COMPLETED = "ProcessCompleted",
}

export const ProcessRecord = ({
  person,
  property,
  process,
  processConfig,
  simple = false,
}: ProcessRecordProps): JSX.Element => {
  const processName = processConfig?.name || process.processName;
  const timeConstraint =
    processConfig?.states[process.state.charAt(0).toLowerCase() + process.state.slice(1)]
      ?.timeConstraint;
  let status;
  if (processConfig) {
    const statesInfo: ProcessStateInfo[] = Object.values(processConfig?.states);
    status = statesInfo.find((item) => item.state === process.state)?.status;
  }

  let daysDiff = 0;
  let hoursLeft = 48;
  if (timeConstraint) {
    daysDiff =
      timeConstraint - differenceInDays(new Date(), parseISO(process.stateStartedAt));
    const completionDate = addDays(parseISO(process.stateStartedAt), timeConstraint);
    hoursLeft = differenceInHours(completionDate, new Date());
  }

  const isClosed =
    status &&
    [
      ProcessStates.PROCESS_CLOSED,
      ProcessStates.PROCESS_CANCELLED,
      ProcessStates.PROCESS_COMPLETED,
    ].includes(status);

  const getState = () => {
    if (process.state === ProcessStates.PROCESS_CANCELLED) {
      return { title: "Cancelled", date: formatDate(process.stateStartedAt) };
    }
    if (process.state === ProcessStates.PROCESS_CLOSED) {
      return { title: "Closed", date: formatDate(process.stateStartedAt) };
    }
    if (process.state === ProcessStates.PROCESS_COMPLETED) {
      return { title: "Completed", date: formatDate(process.stateStartedAt) };
    }
    return {
      title: "Initiated",
      date: formatDate(process.processCreatedAt),
    };
  };

  const state = getState();

  const getStatusColor = () => {
    if (
      status &&
      (
        [ProcessStates.PROCESS_CLOSED, ProcessStates.PROCESS_CANCELLED] as string[]
      ).includes(process.state)
    ) {
      return "orange";
    }
    if (status && process.state === ProcessStates.PROCESS_COMPLETED) {
      return "green";
    }
    if (daysDiff === timeConstraint) {
      return "grey";
    }
    if (hoursLeft >= 48) {
      return "aqua";
    }
    if (hoursLeft > 0 && hoursLeft < 48) {
      return "amber";
    }
    return "red";
  };

  return (
    <Tr className="govuk-table__row process-record">
      <Td className="process-record__item">
        <Link
          as={RouterLink}
          to={`/person/${person.id}`}
          variant="link"
          target="_blank"
          className="lbh-heading-h4"
          data-testid="person-link"
        >
          {person.fullName}
        </Link>
        {property && property.id && property.address && (
          <Link
            className="--address"
            as={RouterLink}
            to={`/property/${property.id}`}
            variant="link"
            target="_blank"
            data-testid="property-link"
          >
            {property.address}
          </Link>
        )}
      </Td>
      <Td className="process-record__item --processName">
        <Link
          as={RouterLink}
          to={`/processes/${
            Number.isInteger(process.processName)
              ? ProcessName[process.processName]
              : process.processName
          }/${process.id}`}
          variant="link"
          target="_blank"
          className="lbh-heading-h4"
        >
          {processName}
        </Link>
      </Td>
      {!simple && (
        <Td className="process-record__item">
          <Text>{process.patchAssignment?.patchName}</Text>
        </Td>
      )}
      <Td className="process-record__item">
        <Text className="title">{state.title}</Text>
        <Text className="createdAt">{state.date}</Text>
      </Td>
      {!simple && (
        <Td className="process-record__item">
          <Text>
            {isClosed || !timeConstraint
              ? "-"
              : `${Math.abs(daysDiff)} days ${daysDiff > 0 ? "left" : "overdue"}`}
          </Text>
        </Td>
      )}
      <Td className="process-record__item --status">
        <span className={`govuk-tag lbh-tag box ${getStatusColor()}`}>
          {status || ""}
        </span>
        {!simple && (
          <Text size="sm">
            {!isClosed &&
              timeConstraint &&
              `${Math.abs(daysDiff)} days ${daysDiff > 0 ? "left" : "overdue"}`}
          </Text>
        )}
      </Td>
    </Tr>
  );
};
