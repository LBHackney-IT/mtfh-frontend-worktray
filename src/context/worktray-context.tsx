import React, {
  Dispatch,
  FC,
  Reducer,
  createContext,
  useContext,
  useEffect,
  useMemo,
  useReducer,
} from "react";
import { useHistory, useLocation } from "react-router-dom";

import { stringify } from "query-string";

import { AxiosSWRError, useAxiosSWR } from "@mtfh/common/lib/http";

import { config } from "../services";
import {
  LimitOptions,
  OrderByOptions,
  Process,
  ProcessSortOptions,
  TimePeriodOptions,
  WorktrayFilterOptions,
} from "../types";

interface WorktrayResults {
  results: {
    processes: Process[];
  };
  total: number;
}

export type WorktrayState = {
  page: number;
  pageSize: LimitOptions;
  timePeriod: TimePeriodOptions;
  sort: ProcessSortOptions;
  order: OrderByOptions;
  patch?: string;
  areaId?: string | null;
  processNames?: string;
  status?: string;
  results?: Process[];
  total?: number;
  error?: AxiosSWRError;
};

export type WorktrayActions =
  | {
      type: "PAGE";
      payload: number;
    }
  | {
      type: "LIMIT";
      payload: LimitOptions;
    }
  | {
      type: "TIME_PERIOD";
      payload: TimePeriodOptions;
    }
  | {
      type: "SORT";
      payload: ProcessSortOptions;
    }
  | {
      type: "FILTER";
      payload: {
        type: WorktrayFilterOptions;
        payload: string;
      };
    }
  | {
      type: "ORDER";
      payload: OrderByOptions;
    };

export interface WorktrayProviderProps {
  initial: Partial<WorktrayState>;
}

const getInitialState = (
  options: WorktrayProviderProps["initial"] = {},
): WorktrayState => {
  return {
    page: options.page || 1,
    pageSize: options.pageSize || LimitOptions.SMALL,
    timePeriod: options.timePeriod || TimePeriodOptions.DAYS_30,
    sort: options.sort || ProcessSortOptions.STATUS,
    order: options.order || OrderByOptions.ASC,
    processNames: options.processNames || undefined,
    patch: options.patch || undefined,
    areaId: options.areaId || undefined,
    status: options.status || undefined,
  };
};

export const WorktrayContext = createContext<{
  state: WorktrayState;
  dispatch: Dispatch<WorktrayActions>;
}>({
  state: getInitialState(),
  dispatch: () => {},
});

export const WorktrayProvider: FC<WorktrayProviderProps> = ({ children, initial }) => {
  const reducer: Reducer<WorktrayState, WorktrayActions> = (state, action) => {
    switch (action.type) {
      case "PAGE": {
        return {
          ...state,
          page: action.payload,
        };
      }
      case "LIMIT": {
        if (Object.values(LimitOptions).includes(action.payload)) {
          return {
            ...state,
            pageSize: action.payload,
          };
        }
        return state;
      }
      case "TIME_PERIOD": {
        if (Object.values(TimePeriodOptions).includes(Number(action.payload))) {
          return {
            ...state,
            timePeriod: action.payload,
          };
        }
        return state;
      }
      case "SORT": {
        if (Object.values(ProcessSortOptions).includes(action.payload)) {
          return {
            ...state,
            sort: action.payload,
          };
        }
        return state;
      }
      case "ORDER": {
        if (Object.values(OrderByOptions).includes(action.payload)) {
          return {
            ...state,
            order: action.payload,
          };
        }
        return state;
      }
      case "FILTER": {
        return {
          ...state,
          [action.payload.type]: action.payload.payload,
        };
      }
      default: {
        return state;
      }
    }
  };

  const initialState = getInitialState(initial);
  const [state, dispatch] = useReducer(reducer, initialState);
  const { ...query } = state;

  const { data, error } = useAxiosSWR<WorktrayResults>(
    `${config.searchApiUrl}/search/processes?${stringify(
      {
        searchText: query.patch,
        page: query.page,
        pageSize: query.pageSize,
        timePeriod: query.timePeriod,
        processNames: query.processNames,
        status: query.status,
        sortBy: query.sort,
        isDesc: query.order === "asc",
      },
      {
        skipEmptyString: true,
      },
    )}`,
    {
      refreshWhenHidden: false,
      revalidateOnFocus: false,
    },
  );

  useEffect(() => {
    if (data?.total && data.total > 0) {
      const totalPages = Math.ceil(data.total / state.pageSize);
      if (state.page > totalPages) {
        dispatch({ type: "PAGE", payload: totalPages });
      }
    }
  }, [state.page, data?.total, state.pageSize]);

  const value = useMemo(
    () => ({
      state: {
        ...state,
        results: data?.results.processes,
        total: data?.total,
        error,
      },
      dispatch,
    }),
    [state, data, error, dispatch],
  );

  return <WorktrayContext.Provider value={value}>{children}</WorktrayContext.Provider>;
};

export const PersistWorktrayContextURL = ({
  sessionKey,
}: {
  sessionKey?: string;
}): null => {
  const {
    state: { page, pageSize, order, sort, timePeriod, patch, processNames, status },
  } = useContext(WorktrayContext);
  const { push } = useHistory();

  useEffect(() => {
    const query = new URLSearchParams();
    if (page > 1) {
      query.set("p", String(page));
    }

    if (pageSize !== LimitOptions.SMALL) {
      query.set("l", String(pageSize));
    }

    query.set("t", String(timePeriod));

    if (order && order !== OrderByOptions.ASC) {
      query.set("o", order);
    }

    query.set("sort", String(sort));

    if (patch) {
      query.set("patch", patch);
    }

    if (processNames) {
      query.set("processNames", processNames);
    }

    if (status) {
      query.set("status", status);
    }

    const path = `?${query.toString()}`;
    push(path);

    if (sessionKey) {
      window.sessionStorage.setItem(sessionKey, path);
    }
  }, [
    page,
    pageSize,
    order,
    sort,
    sessionKey,
    push,
    timePeriod,
    patch,
    processNames,
    status,
  ]);

  return null;
};

export const WorktrayURLProvider: FC<{
  sessionKey?: string;
  patchId?: string | null;
  areaId?: string | null;
}> = ({ children, sessionKey, patchId, areaId }): JSX.Element => {
  const { search } = useLocation();
  const query = new URLSearchParams(search);
  const page = Number(query.get("p") || 1);
  const pageSize = Number(query.get("l") || 10);
  const timePeriod = (query.get("t") || "") as TimePeriodOptions;
  const patch = query.get("patch") || patchId || "";
  const processNames = query.get("processNames") || "";
  const status = query.get("status") || "";
  const sortBy = query.get("sort");
  const orderBy = query.get("o");

  const sort = useMemo(() => {
    if (sortBy === "name") {
      return ProcessSortOptions.NAME;
    }
    return ProcessSortOptions.STATUS;
  }, [sortBy]);

  const order = useMemo(() => {
    if (orderBy === "desc") {
      return OrderByOptions.DESC;
    }
    return OrderByOptions.ASC;
  }, [orderBy]);

  return (
    <WorktrayProvider
      initial={{
        page,
        sort,
        order,
        pageSize,
        timePeriod,
        patch,
        areaId,
        processNames,
        status,
      }}
    >
      {children}
      <PersistWorktrayContextURL sessionKey={sessionKey} />
    </WorktrayProvider>
  );
};
