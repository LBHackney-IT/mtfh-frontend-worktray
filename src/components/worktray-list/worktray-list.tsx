import React, { useCallback, useContext } from "react";

import {
  Center,
  ErrorSummary,
  Spinner,
  Table,
  Tbody,
  Text,
  Th,
  Thead,
  Tr,
} from "@mtfh/common/lib/components";
import { processes } from "@mtfh/processes";

import { WorktrayContext } from "../../context/worktray-context";
import { locale } from "../../services";
import { OrderByOptions, ProcessName, ProcessSortOptions } from "../../types";
import { WorktrayPagination } from "../worktray-pagination";
import { PersonProcessRecord } from "./person-record";
import { TenureProcessRecord } from "./tenure-record";

export const WorktrayList = (): JSX.Element => {
  const {
    state: { results, error, order, sort },
    dispatch,
  } = useContext(WorktrayContext);

  const dispatchSort = useCallback(
    (sortOption) => {
      dispatch({ type: "SORT", payload: sortOption });
      dispatch({
        type: "ORDER",
        payload:
          sort === sortOption && order === OrderByOptions.ASC
            ? OrderByOptions.DESC
            : OrderByOptions.ASC,
      });
    },
    [dispatch, sort, order],
  );

  const processRecordComponents = {
    tenure: TenureProcessRecord,
    person: PersonProcessRecord,
  };

  if (error) {
    return (
      <ErrorSummary
        id="worktray-result-error"
        title={locale.errors.unableToFetchRecord}
        description={locale.errors.unableToFetchRecordDescription}
      />
    );
  }

  if (results === undefined) {
    return (
      <Center className="mtfh-worktray__loading">
        <Spinner />
      </Center>
    );
  }

  if (results?.length === 0) {
    return <Text>{locale.views.worktray.noWorktrayResults}</Text>;
  }

  return (
    <div>
      <div className="worktray-list">
        <Table>
          <Thead>
            <Tr>
              <Th>
                <button
                  className="select-all lbh-link lbh-link--no-visited-state"
                  onClick={() => dispatchSort(ProcessSortOptions.NAME)}
                >
                  Name / Address
                </button>
              </Th>
              <Th>
                <button
                  className="select-all lbh-link lbh-link--no-visited-state"
                  onClick={() => dispatchSort(ProcessSortOptions.PROCESS)}
                >
                  Process
                </button>
              </Th>
              <Th>
                <button
                  className="select-all lbh-link lbh-link--no-visited-state"
                  onClick={() => dispatchSort(ProcessSortOptions.PATCH)}
                >
                  Patch
                </button>
              </Th>
              <Th>
                <button
                  className="select-all lbh-link lbh-link--no-visited-state"
                  onClick={() => dispatchSort(ProcessSortOptions.STATE)}
                >
                  State
                </button>
              </Th>
              <Th>
                <button
                  className="select-all lbh-link lbh-link--no-visited-state"
                  onClick={() => dispatchSort(ProcessSortOptions.TIME_LEFT)}
                >
                  Time left
                  <br />
                  Full process
                </button>
              </Th>
              <Th>
                Process status
                <span id="process-status-info-icon" />
                <div id="process-status-info">
                  <Text className="red">Red represents overdue</Text>
                  <Text className="orange">Orange represents closed case</Text>
                  <Text className="amber">
                    Yellow represents 48 hours warning until overdue
                  </Text>
                  <Text className="green">Green represents completed case</Text>
                  <Text className="aqua">Aqua represents in progress</Text>
                  <Text className="grey">Grey represents new case</Text>
                </div>
              </Th>
            </Tr>
          </Thead>
          <Tbody>
            {results.map((process) => {
              const processConfig =
                processes[
                  Number.isInteger(process.processName)
                    ? ProcessName[process.processName]
                    : process.processName
                ];
              const Result = processRecordComponents[process.targetType];
              return (
                <Result
                  key={process.id}
                  process={process}
                  processConfig={processConfig}
                />
              );
            })}
          </Tbody>
        </Table>
        <WorktrayPagination />
      </div>
    </div>
  );
};
