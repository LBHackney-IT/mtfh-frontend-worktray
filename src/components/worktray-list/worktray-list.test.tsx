import React from "react";

import { mockPersonV1, mockProcessV1, render, server } from "@hackney/mtfh-test-utils";
import { fireEvent, screen, waitForElementToBeRemoved } from "@testing-library/react";
import { rest } from "msw";

import { WorktrayProvider, WorktrayURLProvider } from "../../context/worktray-context";
import { locale } from "../../services";
import { WorktrayList } from "./worktray-list";

const mockWorktrayResults = [
  {
    ...mockProcessV1,
    processName: "soletojoint",
    targetType: "tenure",
    currentState: {
      createdAt: "2022-01-09T00:00:00.000Z",
    },
    relatedEntities: [
      {
        id: mockPersonV1.id,
        description: `${mockPersonV1.firstName} ${mockPersonV1.surname}`,
        targetType: "person",
        subType: "",
      },
    ],
  },
];

describe("worktray-list-component", () => {
  beforeEach(() => {
    jest.resetModules();
  });

  test("it renders correctly", async () => {
    server.use(
      rest.get("/api/worktray", (req, res, ctx) => {
        return res(
          ctx.status(200),
          ctx.json({
            results: mockWorktrayResults,
          }),
        );
      }),
    );
    const { container } = render(
      <WorktrayProvider initial={{}}>
        <WorktrayList />
      </WorktrayProvider>,
    );
    await waitForElementToBeRemoved(screen.queryByText(/Loading/));
    expect(container).toMatchSnapshot();
  });

  test("it renders correctly when no results", async () => {
    server.use(
      rest.get("/api/worktray", (req, res, ctx) => {
        return res(ctx.status(200), ctx.json({ results: [] }));
      }),
    );
    const { container } = render(
      <WorktrayProvider initial={{}}>
        <WorktrayList />
      </WorktrayProvider>,
    );
    await waitForElementToBeRemoved(screen.queryByText(/Loading/));
    expect(container).toMatchSnapshot();
  });

  test("it renders error", async () => {
    server.use(
      rest.get("/api/worktray", (req, res, ctx) => {
        return res(ctx.status(500), ctx.json({}));
      }),
    );
    render(
      <WorktrayProvider initial={{}}>
        <WorktrayList />
      </WorktrayProvider>,
    );
    await waitForElementToBeRemoved(screen.queryByText(/Loading/));
    await expect(
      screen.findByText(locale.errors.unableToFetchRecord),
    ).resolves.toBeInTheDocument();
  });

  [
    ["Name / Address", "name"],
    ["Process", "process"],
    ["Patch", "patch"],
    ["State", "state"],
    [/Time left/, "time_left"],
  ].forEach((entry) => {
    test(`Table headers dispatch actions for ${entry[0]}`, async () => {
      server.use(
        rest.get("/api/worktray", (req, res, ctx) => {
          return res(ctx.status(200), ctx.json({ results: mockWorktrayResults }));
        }),
      );
      render(
        <WorktrayURLProvider sessionKey="test">
          <WorktrayList />
        </WorktrayURLProvider>,
      );

      await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));
      fireEvent.click(screen.getByText(entry[0]));
      expect(window.sessionStorage.getItem("test")).toBe(`?t=30&sort=${entry[1]}`);

      await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));
      fireEvent.click(screen.getByText(entry[0]));
      expect(window.sessionStorage.getItem("test")).toBe(`?t=30&o=desc&sort=${entry[1]}`);
    });
  });
});
