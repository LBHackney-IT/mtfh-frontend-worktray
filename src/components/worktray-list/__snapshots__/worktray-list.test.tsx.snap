// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`worktray-list-component it renders correctly 1`] = `
<div>
  <div>
    <div
      class="worktray-list"
    >
      <table
        class="govuk-table lbh-table"
      >
        <thead
          class="govuk-table__head"
        >
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__cell govuk-table__header"
            >
              <button
                class="select-all lbh-link lbh-link--no-visited-state"
              >
                Name / Address
              </button>
            </th>
            <th
              class="govuk-table__cell govuk-table__header"
            >
              <button
                class="select-all lbh-link lbh-link--no-visited-state"
              >
                Process
              </button>
            </th>
            <th
              class="govuk-table__cell govuk-table__header"
            >
              <button
                class="select-all lbh-link lbh-link--no-visited-state"
              >
                Patch
              </button>
            </th>
            <th
              class="govuk-table__cell govuk-table__header"
            >
              <button
                class="select-all lbh-link lbh-link--no-visited-state"
              >
                State
              </button>
            </th>
            <th
              class="govuk-table__cell govuk-table__header"
            >
              <button
                class="select-all lbh-link lbh-link--no-visited-state"
              >
                Time left
                <br />
                Full process
              </button>
            </th>
            <th
              class="govuk-table__cell govuk-table__header"
            >
              Process status
              <span
                id="process-status-info-icon"
              />
              <div
                id="process-status-info"
              >
                <p
                  class="lbh-body-m red"
                >
                  Red represents overdue
                </p>
                <p
                  class="lbh-body-m orange"
                >
                  Orange represents closed case
                </p>
                <p
                  class="lbh-body-m amber"
                >
                  Yellow represents 48 hours warning until overdue
                </p>
                <p
                  class="lbh-body-m green"
                >
                  Green represents completed case
                </p>
                <p
                  class="lbh-body-m aqua"
                >
                  Aqua represents in progress
                </p>
                <p
                  class="lbh-body-m grey"
                >
                  Grey represents new case
                </p>
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="govuk-table__body"
        >
          <tr
            class="govuk-table__row govuk-table__row process-record"
          >
            <td
              class="govuk-table__cell process-record__item"
            >
              <a
                class="govuk-link lbh-link lbh-link--no-visited-state lbh-heading-h4"
                data-testid="person-link"
                href="/person/6fbe024f-2316-4265-a6e8-d65a837e308a"
                target="_blank"
              >
                Haylee Macejkovic
              </a>
            </td>
            <td
              class="govuk-table__cell process-record__item --processName"
            >
              <a
                class="govuk-link lbh-link lbh-link--no-visited-state lbh-heading-h4"
                href="/processes/soletojoint/6fbe024f-2316-4265-a6e8-d65a837e308a"
                target="_blank"
              >
                Sole to Joint
              </a>
            </td>
            <td
              class="govuk-table__cell process-record__item"
            >
              <p
                class="lbh-body-m"
              >
                CP7
              </p>
            </td>
            <td
              class="govuk-table__cell process-record__item"
            >
              <p
                class="lbh-body-m title"
              >
                Initiated
              </p>
              <p
                class="lbh-body-m createdAt"
              >
                20/08/2022
              </p>
            </td>
            <td
              class="govuk-table__cell process-record__item"
            >
              <p
                class="lbh-body-m"
              >
                8 days left
              </p>
            </td>
            <td
              class="govuk-table__cell process-record__item --status"
            >
              <span
                class="govuk-tag lbh-tag box aqua"
              >
                Request supporting documents
              </span>
              <p
                class="lbh-body-s"
              >
                8 days left
              </p>
            </td>
          </tr>
          <tr
            class="govuk-table__row govuk-table__row process-record"
          >
            <td
              class="govuk-table__cell process-record__item"
            >
              <a
                class="govuk-link lbh-link lbh-link--no-visited-state lbh-heading-h4"
                data-testid="person-link"
                href="/person/6fbe024f-2316-4265-a6e8-d65a837e308a"
                target="_blank"
              >
                Haylee Macejkovic
              </a>
            </td>
            <td
              class="govuk-table__cell process-record__item --processName"
            >
              <a
                class="govuk-link lbh-link lbh-link--no-visited-state lbh-heading-h4"
                href="/processes/soletojoint/6fbe024f-2316-4265-a6e8-d65a837e308a"
                target="_blank"
              >
                Sole to Joint
              </a>
            </td>
            <td
              class="govuk-table__cell process-record__item"
            >
              <p
                class="lbh-body-m"
              >
                CP7
              </p>
            </td>
            <td
              class="govuk-table__cell process-record__item"
            >
              <p
                class="lbh-body-m title"
              >
                Initiated
              </p>
              <p
                class="lbh-body-m createdAt"
              >
                20/08/2022
              </p>
            </td>
            <td
              class="govuk-table__cell process-record__item"
            >
              <p
                class="lbh-body-m"
              >
                8 days left
              </p>
            </td>
            <td
              class="govuk-table__cell process-record__item --status"
            >
              <span
                class="govuk-tag lbh-tag box aqua"
              >
                Request supporting documents
              </span>
              <p
                class="lbh-body-s"
              >
                8 days left
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`worktray-list-component it renders correctly when no results 1`] = `
<div>
  <p
    class="lbh-body-m"
  >
    No worktray results
  </p>
</div>
`;
